src/search/landmarks/landmark_graph.cc

```diff
@@ -118,43 +101,28 @@ bool LandmarkGraph::contains_landmark(const FactPair &lm) const {
     return contains_simple_landmark(lm) || contains_disjunctive_landmark(lm);
 }
 
-LandmarkNode &LandmarkGraph::add_simple_landmark(const FactPair &lm) {
-    assert(!contains_landmark(lm));
-    vector<FactPair> facts{lm};
-    unique_ptr<LandmarkNode> new_node =
-        utils::make_unique_ptr<LandmarkNode>(facts, false, false);
-    LandmarkNode *new_node_p = new_node.get();
-    nodes.push_back(move(new_node));
-    simple_landmarks_to_nodes.emplace(lm, new_node_p);
-    return *new_node_p;
-}
-
-LandmarkNode &LandmarkGraph::add_disjunctive_landmark(const set<FactPair> &lm) {
-    assert(all_of(lm.begin(), lm.end(), [&](const FactPair &lm_fact) {
+LandmarkNode &LandmarkGraph::add_landmark(Landmark &&landmark) {
+    assert(all_of(landmark.facts.begin(), landmark.facts.end(), [&](const FactPair &lm_fact) {
                       return !contains_landmark(lm_fact);
                   }));
-    vector<FactPair> facts(lm.begin(), lm.end());
+    // TODO: This turned out to be very ugly with the creation of the unique
+    //  pointer which moves the Landmark object, then the getting of the raw
+    //  pointer and then the re-getting of the Landmark object.
```

Comment:
Wrong comment format (but I guess you want to remove the TODO before merging?)

Regarding the content of the TODO, I think the code looks much nicer than before. Getting the raw pointer is indeed somewhat ugly but it was the same before and I can't think of a better way other than const reference but I guess that won't work here.
I'm not too good with move semantics so I can't really judge that part but otherwise the code looks good to me.

Reply:
I removed the TODO.